src/types/sig.rs

```diff
@@ -390,21 +390,15 @@ mod crypto {
     }
 
     fn loadsig(path: &str) -> Signature {
-        let mut sig: Signature;
-        let buf: &mut [u8];
-
-        unsafe {
-            sig = std::mem::MaybeUninit::uninit().assume_init();
-            buf = std::slice::from_raw_parts_mut(
-                &mut sig as *mut _ as *mut u8,
-                std::mem::size_of_val(&sig),
-            );
-        }
+        let mut sig = std::mem::MaybeUninit::<Signature>::uninit();
```

Comment:
Add a comment explaining a) why this is safe ("..because we don't use it before writing..") and b) why the original code was screwing up ("don't do init after doing uninit" or whatever it was...)